### PR TITLE
LPD-39430 Add support for Knowledge base asset to have publication timeline history

### PR DESCRIPTION
**Pull request by ctang3**
Not sure how to deal with these SF errors for the changes in CTEntryLocalServiceImpl since afaik there isn't another way to do it. Any insight @david-truong?

```
1: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 345 (Checkstyle:ChainingCheck)
2: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 347 (Checkstyle:ChainingCheck)
3: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 349 (Checkstyle:ChainingCheck)
4: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 351 (Checkstyle:ChainingCheck)
5: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 359 (Checkstyle:ChainingCheck)
6: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 361 (Checkstyle:ChainingCheck)
7: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 363 (Checkstyle:ChainingCheck)
8: Avoid chaining on Type Cast: /home/cheryl/Work/Repos/liferay-portal/modules/apps/change-tracking/change-tracking-service/src/main/java/com/liferay/change/tracking/service/impl/CTEntryLocalServiceImpl.java 364 (Checkstyle:ChainingCheck)
```

### CODE REVIEW

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:relevant

**Comment by ctang3** [comment]
ci:test:publications

**Comment by ctang3** [comment]
ci:test:relevant

File: modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java [hunk 1]
```diff
@@ -632,6 +632,16 @@ private void _getTimelineData(
 				className = Layout.class.getName();
 				classPK = layout.getPlid();
 			}
+			else {
+				String namespace = _portal.getPortletNamespace(
+					httpServletRequest.getParameter("p_p_id"));
```
Comment (david-truong) [inline modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java:637]: what's different about knowledge base that we need to add this?

**Comment by david-truong** [comment]
you will need wait for https://github.com/liferay-site-management/liferay-portal/pull/1524 and rebase on that.  we should have never messed with the default fetch method this way

**Pull request closed without merging**